docs/semgrep-supply-chain/malicious-dependencies.md

```diff
@@ -0,0 +1,74 @@
+---
+slug: malicious-dependencies
+title: Detect and remove malicious dependencies
+hide_title: true
+description: Learn how Semgrep detects malicious dependencies and enable malicious dependency detection in your Supply Chain scans.
+tags:
+  - Semgrep Supply Chain
+---
+
+# Detect and remove malicious dependencies
+
+:::info
+This feature is in private beta. To join, reach out to [support](/support).
+:::
+
+**Malicious dependencies** are dangerous packages, or dangerous versions of packages, that are designed to compromise systems. These threats include packages that have always been malicious, such as typo-squatting attacks, or packages that become malicious after an attacker compromises a maintainer or injects harmful code. They are also known as malware.
+
+Semgrep is able to detect malicious dependencies in your projects and in pull requests (PRs) or merge requests (MRs).
+
+This feature is enabled after opting in to the beta program.
+
+## Supported package managers and sources of information
+
+Semgrep ingests data from Open Source Vulnerabilities ([<i class="fas fa-external-link fa-xs"></i>  osv.dev](https://osv.dev/)).
+
+The following table lists the languages for which Supply Chain can detect malicious dependencies.
+
+| Language | Package manager or ecosystem |
+| :---- | :---- |
+| C\# | NuGet |
+| Dart | Pub |
+| Elixir | Hex |
+| Go | `go.mod` |
+| Java | Maven |
+| Python | PyPi |
+| Ruby | RubyGems |
+| Scala | Maven |
+| Swift | SwiftPM |
+| PHP | \-- |
+| Rust | \-- |
+
+## About malicious dependency findings
+
+Malicious dependency findings are treated as **critical severity** findings.
+
+If you have set up your Supply Chain policies to block with these conditions, malicious dependency findings block a PR or MR in the same way as any other Supply Chain finding.
+
+<!--  No way to do this currently
+## Enable or disable malicious dependency detection
+
+1. Click Settings \> …
+-->
+
+## View malicious dependencies
+
+Malicious dependencies appear in [**Supply Chain > Vulnerabilities**](https://semgrep.dev/orgs/-/supply-chain/vulnerabilities?primary=true&tab=open&last_opened=All+time), alongside other Supply Chain findings. They are denoted by the **Malicious** badge.
+
+![Malicious dependency finding with badge](/img/findings-maldeps.png)
+_**Figure**. A malicious dependency finding._
+
+Use the **Malicious dependencies <i class="fa-solid fa-toggle-large-on"></i> toggle** to filter for malicious dependencies detected in your projects. Ensure that you don't have other filters enabled as this may inadvertently hide findings.
```

Comment:
Do you know what use cases would result in mal deps hidden by other filters (this question is just for my own info)?

Reply:
I apologize for the late answer - basically, the only **attribute** for a mal dep finding is uh, critical severity. so, with the mal deps filter on, if you add ANY OTHER FILTER, epss (any value, whether high, medium, low, etc), reachability (any value), etc, it shows nothing because the filter is additive.
![Snag_2d6e9b8](https://github.com/user-attachments/assets/28927b9a-fe33-4d6d-9820-0155f6d5bc66)